privacyidea/api/auth.py

```diff
@@ -114,7 +114,7 @@ def before_request():
         # On endpoints like /auth/rights, this is not available
         loginname, realm = split_user(username)
         # overwrite the split realm if we have a realm parameter. Default back to default_realm
-        realm = getParam(request.all_data, "realm", default=realm) or realm or get_default_realm()
+        realm = getParam(request.all_data, "realm", default=realm) or get_default_realm()
```

Comment:
wouldn't a a 

    realm = realm.lower()


be sufficient?

Reply:
here we have no problems with the case of the realm. The realm will be converted to lower-case automatically when we create the `User` object.